File: C++/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P03_Vacation.cpp
#include <iostream>
#include <algorithm>
#include <string>
using namespace std;

int main() {

	double priceVacation, availableMoney;
	cin >> priceVacation >> availableMoney;

	int savedCounter = 0;
	int spentCounter = 0;

	string command;
	double currMoney = 0;
	while (availableMoney < priceVacation) {
		cin >> command;
		cin >> currMoney;

		if (command == "save") {
			availableMoney += currMoney;
			spentCounter = 0;
		}
		else if (command == "spend") {
			availableMoney = max(availableMoney - currMoney, 0.00);
			spentCounter++;
		}

		savedCounter++;

		if (spentCounter == 5) {
			cout << "You can't save the money." << endl;
			cout << savedCounter << endl;

			return 0;
		}
	}

	cout << "You saved the money for " << savedCounter << " days." << endl;

	return 0;
}

File: Python/M01_ProgrammingBasics/L05_WhileLoop/Exercises/Solutions/P03_Vacation.py
money_needed = float(input())
money_have = float(input())
amount = 0
number_spend = 0
days = 0

while True:
    act = input()
    amount = float(input())
    days += 1

    if act == "spend":
        number_spend += 1
        money_have -= amount
        if money_have <= 0:
            money_have = 0
    else:
        number_spend = 0

    if number_spend == 5:
        print(f"You can't save the money.")
        print(f"{days}")
        break

    if act == "save":
        money_have += amount

    if money_have >= money_needed:
        print(f"You saved the money for {days} days.")
        break
